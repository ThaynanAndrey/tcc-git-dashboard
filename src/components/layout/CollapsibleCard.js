import React from 'react';

/**
 * Stateless component with the function of presenting a Collapsible Card.
 * 
 * @author Thaynan Nunes
 */
const CollapsibleCard = ({ children, isActive }) => (
    <li className={isActive ? "active" : ""}>
        { children }
    </li>
);

export default CollapsibleCard;