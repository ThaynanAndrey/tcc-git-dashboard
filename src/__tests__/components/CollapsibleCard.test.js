import React from 'react';
import { shallow, configure } from "enzyme";
import toJson from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';

import CollapsibleCard from '../../components/layout/CollapsibleCard';

configure({ adapter: new Adapter() });

describe('Test CollapsibleCard', () => {
  it('renders correctly a collapsible without props isActive', () => {
    const tree = shallow(
        <CollapsibleCard />
    );

    expect(toJson(tree)).toMatchSnapshot();
  });

  it('renders correctly a collapsible with props isActive', () => {
    const tree = shallow(
        <CollapsibleCard isActive={true}/>
    );

    expect(toJson(tree)).toMatchSnapshot();
  });

  it('the component should have class active when props isActive is true', () => {
    const tree = shallow(
        <CollapsibleCard isActive={true}/>
    );

    expect(tree.find(".active").exists()).toBeTruthy();
  });

  it('the component shouldn\'t have class active when props isActive is false', () => {
    const tree = shallow(
        <CollapsibleCard isActive={false}/>
    );

    expect(tree.find(".active").exists()).toBeFalsy();
  });
});