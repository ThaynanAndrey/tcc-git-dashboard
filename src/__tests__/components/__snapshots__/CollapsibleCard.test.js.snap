// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CollapsibleCard renders correctly a collapsible with props isActive 1`] = `
<li
  className="active"
/>
`;

exports[`Test CollapsibleCard renders correctly a collapsible without props isActive 1`] = `
<li
  className=""
/>
`;
